engine/lite/components/base/src/main/scala/pl/touk/nussknacker/engine/lite/components/ForEachTransformer.scala

```diff
@@ -29,8 +29,9 @@ class ForEachTransformerComponent(elements: LazyParameter[java.util.Collection[A
     val interpreter = context.interpreter.syncInterpretationFunction(elements)
     (ctx: Context) => {
       val partsToRun = interpreter(ctx)
-      val partsToInterpret = partsToRun.asScala.toList.map { partToRun =>
-        ctx.withVariable(outputVariable, partToRun)
+      val partsToInterpret = partsToRun.asScala.toList.zipWithIndex.map { case (partToRun, index) =>
+        val parentContextId = ctx.parentContext.fold("")(c => s"${c.id}_")
+        ctx.withVariable(outputVariable, partToRun).copy(id=s"${ctx.id}_$parentContextId$index")
```

Comment:
I've taken a look into `ContextIdGenerator` and generated context id is dash saparated. Maybe would be better to separate index also with dash? WDYT?